inference/core/cache/serializers.py

```diff
@@ -52,10 +52,19 @@ def build_condensed_response(responses):
         if not getattr(response, "predictions", None):
             continue
         try:
-            predictions = [
-                {"confidence": pred.confidence, "class": pred.class_name}
-                for pred in response.predictions
-            ]
+            predictions = []
+            for pred in response.predictions:
+                if type(pred) == str:
+                    predictions.append(
+                        {"class": pred, "confidence": request.confidence}
```

Comment:
this is not that clear for me - why to put a request confidence into predicted class name? That's the model which dictates confidence of prediction, request do only have threshold

Reply:
That was mistake, good catch